addons/website/static/src/scss/website.scss

```diff
@@ -1187,6 +1187,27 @@ header {
         padding-left: $grid-gutter-width / 2;
         padding-right: $grid-gutter-width / 2;
     }
+
+    // The design of some of the new mega menu templates required a column with
+    // an overlay that would overflow until the end of the container. For
+    // technical reasons, this overlay has to be hardcoded here. This specific
+    // css can always be moved back to the XXX.scss appropriate files.
+    .s_mega_menu_gray_area:last-child {
+        z-index: 1;
+
+        &:before {
+            content: '';
+            display: block;
+            width: 100vw;
+            height: 100%;
+            position: absolute;
+            left: 0;
+            top: 0;
+            z-index: -1;
+            pointer-events: none;
+            background: rgba(0, 0, 0, .05);
+        }
+    }
```

Comment:
Sorry I changed my mind about this (note that originally I asked to keep a comment in the XXX.scss file to indicate that this rule exist so it would not be forgotten if we ever do a XXX+1.scss file which does not need it anymore). I don't like "ugly" code to be standard. Instead, make a [placeholder rule](https://sass-lang.com/documentation/at-rules/extend#placeholder-selectors) and extend it in the appropriate XXX.scss templates). That way we still share the code, we still create only one rule about this but we keep it specific to each template.

At some point we should disable the scss of mega menu templates which are not used by the user (like we with header templates) so it will make even more sense to do it this way.